### ISSUE TITLE
2 instances: pause breaks sound

### ISSUE THREAD

**Issue by p3pp0** [popcornmix/omxplayer#317]
Hey Guys,

i tried to get 2 videos play seamlesly (i know the loop function of omxplayer itself, but it is not useful for my special usecase)
Thus i found a solution to "precache" the second video via pause command, so i cant start it seamplessly after the first ends. Both instances run on different dbus_names, so i can control them via 2 different dbuscontrol.sh files

I now get the following issue:
If i start and pause immediatelly instance 2 while instance 1 is playing, everything works fine, but:
sound of instance 1 is stopping and added to instance 2, when unpaused

I hope u understand my issue, feel free to ask if not - sorry for bad english ;)

Thx a lot!

**Comment by subogero** [comment]
I'm struggling with exactly the same issue in `omxd` which does this precaching for low-gap playback. But music is fine. It's a video specific problem.

**Comment by p3pp0** [comment]
i dont understand exactly... did you say that no-gap playback is working with audio only? What can we do? :)

**Comment by subogero** [comment]
No. omxd is a playlist daemon and always runs 2 omxplayer instances. One is playing, the other is precached (paused at the beginning). When the current track is finished, I just send a `p` to the precached one, so there is a very low gap between the tracks. 

What I'm saying is, that when I'm playing audio tracks only, this muting does not happen, only with videos.

**Comment by popcornmix** [comment]
Unfortunately the audio_render openmax component implements pause by halting audio dma.
This means you cannot pause an audio stream and have another audio stream continue.

I'm afraid I've got to say not supported for now. It would take quite a big architectural change to resolve this, which won't happen soon.

**Comment by p3pp0** [comment]
thanks for your reply
Popcorn: do you see any other chance to seamlessly play 2 videos without using the integrated loop function?

background is, that my playlist needs to be updated dynamically, so i cannot call use integrated loop function

**Comment by popcornmix** [comment]
How seamless?
I'm sure video can be switched from one file to another (through destroying and recreating omx components) with less than a 1 second gap.

You would need to edit omxplayer source.

If you wanted real seamless switching, then a scheme more like Kodi uses when using dvdplayer would be needed. Get the arm in charge of scheduling video and audio.
mmal video decode component sends opaque buffers for each decoded frame.
mmal video renderer is driven by arm with these opaque pointers. 
That separation means no need to close down pipeline when codec/resolution changes.

A second mmal video decode component can be started before the first video file finishes so you can switch instantly.

Audio is much easier if controlled by arm. Resample to a common format and mix on arm (or you could use the audio mixer component).

Performance is not quite so good with this setup, but moderate bitrate 1080p should be achievable on Pi1 and full BluRay bitrates on Pi2.

**Comment by p3pp0** [comment]
i really need real seamless switching, meaning no black frame between two videos.

i already tried to put the following video on a higher layer and start it when the first video is almost over... works most of the time but not safely, because the time to load a video differs a bit always (may also depend on other things going on on the pi i/o)

the talking about mmal is very new to me cause i am not an expert... can you try to point me to the right direction?

**Comment by p3pp0** [comment]
i also thought about a solution written in javascript/html5... many layers and preloading/pause should work there... but cant image that there is a browser-solution for the pi that is capable of playing 1080i/p videos smoothly

**Comment by popcornmix** [comment]
See:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecMMAL.cpp
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoRenderers/MMALRenderer.cpp

This sort of scheme can handle codec/resolution changing without a glitch.

**Comment by p3pp0** [comment]
i am afraid i am not able to do this with my background... its simply to hard for me :-/

**Comment by niwa3836** [comment]
Hi all, Sadly I've hit the same issue. Start video 1 connected to dbus 1, video + audio via HDMI is fine. They start video 2 connected to dbus 2 and immediately pause it via dbus2 command. Immediately then video 1 keeps playing but audio stops :/ I really need a way of having a very short gap between videos but struggling with the excellent omxplayer. Is there any breakthrough here? I tried the MMAL links above and 1. they dont work now, and 2. maybe a skill too much for me. Is there anyone with good ideas or can generally help please?

**Comment by subogero** [comment]
Everything works fine with the Jack audio output. It's an audio-over-HDMI specific problem.

**Comment by JamesH65** [comment]
Please see raspberrypi/firmware#547 for a possible fix to this issue.
